### DISCUSSION TITLE
Synology DS216j (working)

### DISCUSSION THREAD

**Discussion by transilluminate** [icloud-photos-downloader/icloud_photos_downloader#824] [General]
Hi! Thanks for an amazing piece of software!

Just wanted to share how I got [icloud_photos_downloader](https://github.com/icloud-photos-downloader/icloud_photos_downloader) working on my [Synology DS216j](https://www.synology.com/en-uk/company/news/article/Synology_Announces_DS216j) _"The Best NAS Device of 2015"_ 😂

Firstly with a new installation, I was getting errors attempting to run from the PyPI package.
([...] File "PyInstaller/loader/pyimod02_importers.py", line 385, in exec_module [...])

I attempted to install from source (with Entware), however, even with a working gcc, I needed the header files to install the cryptography package as there was not a pre-compiled wheel for the platform:
(/usr/syno/include/platformconfig.h is missing on this system...)

This version of DSM was 7.1.1-42962-6 and didn't show an update available in DSM Control Panel.

I noticed there was an update on the [Synology site](https://archive.synology.com/download/Os/DSM) and picked [7.2.1-69057 (with Update 1)](https://archive.synology.com/download/Os/DSM/7.2.1-69057-1-NanoPacked). Installed manually via Control Panel, rebooted and tried again. Note: I could also have gone to the [support site](https://www.synology.com/en-global/support/download/DS216j?version=7.2#system) for this.

This time, it worked!

It just needed a temporary directory which can execute files, [see the fix](https://github.com/icloud-photos-downloader/icloud_photos_downloader/issues/764).

This may be useful for the few out there also with the same underpowered, almost 10 year old, little box.

**Comment by AndreyNikiforov** [comment]
@transilluminate Thanks for sharing your experience and solution. PyPi packages have binaries for [arm32v7 chip used in DS216j](https://icloud-photos-downloader.github.io/icloud_photos_downloader/nas.html#running-on-synology-nas), so I would expect installation to work without the need for compilation or cryptography. If that does not work as expected, would be nice to understand why and update documentation.
